### ISSUE TITLE
String enum cases compile to NSStrings in Objective-C header

### ISSUE THREAD

**Issue by vasarhelyia** [JetBrains/kotlin-native#3056]
Hi there,

I created a String enum class in a Kotlin Native project which I'm using in another Kotlin types. See below.

```
enum class StrokeLineJoin(val value: String) {
    MITER("miter"),
    ROUND("round"),
    BEVEL("bevel"),
}

data class StrokeParams(
    val strokeLineJoin: StrokeLineJoin = StrokeLineJoin.MITER
)
```

After building the project for iOS, looking at the generated Objective-C header I can see that every occurrence of the enum type is translated to `NSString`:

```
@property (readonly) NSString *strokeLineJoin __attribute__((swift_name("strokeLineJoin")));
```

As a workaround, I'll make it a basic (non-String) enum and just switch over the cases in the native iOS code, but it would've been nice to have the original idea work. Am I missing something here, or that should've worked?

Thanks!

**Comment by SvyatoslavScherbina** [comment]
I can't reproduce this behaviour.

```Kotlin
enum class StrokeLineJoin(val value: String) {
    MITER("miter"),
    ROUND("round"),
    BEVEL("bevel"),
}

data class StrokeParams(
    val strokeLineJoin: StrokeLineJoin = StrokeLineJoin.MITER
)
```

If I compile this code to framework, I get

```Objective-C
@property (readonly) FrameworkStrokeLineJoin *strokeLineJoin __attribute__((swift_name("strokeLineJoin")));
```
as expected.

Which version of Kotlin do you use?
Could you post the entire header of framework produced by compilation of this code on your side?

**Comment by SvyatoslavScherbina** [comment]
Got no response, closing.

**Comment by benasher44** [comment]
Confirmed this is fixed now — 1.3.70 (at least)
